sourcegraph/sourcegraph
# Setup Wizard: add new setup steps UI

Closes https://github.com/sourcegraph/sourcegraph/issues/46641

This PR adds a new setup wizard stepper UI based on Figma designs [here](https://www.figma.com/file/EuRdjjQF8a9kRJMRce4HyI/UX%3A-Instance-setup-experience?node-id=1160%3A11543&t=CVo1uMo6GOVuhYFq-0), Note that proper dark there support will be addressed later in the upcoming PRs (here is a [comment](https://www.figma.com/file/EuRdjjQF8a9kRJMRce4HyI?node-id=1160:11543#361271311) about this in Figma)

Each step of the wizard has its own URL route, so if you refresh the page, you should see the current step. We sync the last valid active step in the temporary store, so if you were on /setup/<setup step URL route> and open just /setup route, you should be redirected to /setup/<setup step URL route> (the last valid step you were)

## Test plan
- Go to the /setup route and make sure that you see the repositories page UI 
- Turn on the experimental feature flag `setupWizard` in your settings and go to the /setup route. Make sure that you see the first step in the wizard UI
- Check that you can go to the second step by clicking the Next button
- Check that the second route renders a custom Next button, and it should be disabled
- Make sure that if you on the first screen and you haven't seen any further steps in the setup wizard you can't open further steps by going to them by their URLs, you should be redirected to the last valid screen instead.



## App preview:

- [Web](https://sg-web-vk-add-new-setup-steps-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
